app/src/main/java/com/github/se/signify/model/user/UserRepositoryFireStore.kt

```diff
@@ -0,0 +1,194 @@
+        onSuccess()
+      }
+    }
+  }
+
+  override fun getFriendsList(
+      userId: String,
+      onSuccess: (List<String>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val db = FirebaseFirestore.getInstance()
+    val userRef = db.collection(collectionPath).document(userId)
+
+    // Use of addSnapshotListener for instant updates
+    userRef.addSnapshotListener { documentSnapshot, e ->
+      if (e != null) {
+        onFailure(e)
+        return@addSnapshotListener
+      }
+
+      if (documentSnapshot != null && documentSnapshot.exists()) {
+        val friends = documentSnapshot.get(friendsListPath) as? List<String>
+        onSuccess(friends ?: emptyList())
+      } else {
+        onSuccess(emptyList())
+      }
+    }
+  }
+
+  override fun getRequestsFriendsList(
+      userId: String,
+      onSuccess: (List<String>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val db = FirebaseFirestore.getInstance()
+    val userRef = db.collection(collectionPath).document(userId)
+
+    // Use of addSnapshotListener for instant updates
+    userRef.addSnapshotListener { documentSnapshot, e ->
+      if (e != null) {
+        onFailure(e)
+        return@addSnapshotListener
+      }
+
+      if (documentSnapshot != null && documentSnapshot.exists()) {
+        val friendsRequests = documentSnapshot.get(friendRequestsListPath) as? List<String>
+        onSuccess(friendsRequests ?: emptyList())
+      } else {
+        onSuccess(emptyList())
+      }
+    }
+  }
+
+  // Search for a user by their userId
+  override fun getUserById(
+      userId: String,
+      onSuccess: (User) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val userRef = db.collection(collectionPath).document(userId)
+
+    userRef
+        .get()
+        .addOnSuccessListener { document ->
+          if (document.exists()) {
+            try {
+              // Make sure the document is properly converted into the User object
+              val user = document.toObject(User::class.java)
+              if (user != null) {
+                onSuccess(user)
+              } else {
+                onFailure(Exception("Failed to parse user data"))
+              }
+            } catch (e: Exception) {
+              onFailure(e)
+            }
+          } else {
+            onFailure(Exception("User not found"))
+          }
+        }
+        .addOnFailureListener { e -> onFailure(e) }
+  }
+
+  override fun sendFriendRequest(
+      currentUserId: String,
+      targetUserId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val targetUserRef = db.collection(collectionPath).document(targetUserId)
+
+    targetUserRef
+        .update(friendRequestsListPath, FieldValue.arrayUnion(currentUserId))
+        .addOnSuccessListener {
+          onSuccess() // Trigger onSuccess callback
+        }
+        .addOnFailureListener { e ->
+          onFailure(e) // Trigger onFailure callback
+        }
+  }
```

Comment:
The UserRepositoryFireStore class provides various user management methods, including getFriendsList, sendFriendRequest, and acceptFriendRequest. The use of FieldValue.arrayUnion() and FieldValue.arrayRemove() for adding/removing friends or requests is a clean and efficient way to manage these list operations.